index.ts

```diff
@@ -30,25 +30,37 @@ function getJestTransformConfig(
   }
 }
 
+function isEmptyTransformOptions(options: any) {
+  return !(options && Object.keys(options).length)
+}
+
 export = {
   process(src: string, filename: string, jestConfig: any) {
 
     if (/\.(t|j)sx?$/.test(filename)) {
 
-      if (!transformOpts) {
+      if (isEmptyTransformOptions(transformOpts)) {
         let swcOptions = getJestTransformConfig(jestConfig);
 
         if (!swcOptions) {
           const swcrc = path.join(process.cwd(), '.swcrc')
           swcOptions = fs.existsSync(swcrc) ? JSON.parse(fs.readFileSync(swcrc, 'utf-8')) as Options : {}
         }
 
-        // set(swcOptions, 'module.type', 'commonjs')
+
         set(swcOptions, 'jsc.transform.hidden.jest', true)
 
         transformOpts = swcOptions
       }
 
+
+      const parserConfig = transformOpts.jsc?.parser;
+
+      // typescript: `tsx` will be modified if file extension is `ts`.
+      if (parserConfig?.syntax === 'typescript') {
```

Comment:
As we pass filename, we can blindly set `tsx: true`. Rust code will disable `tsx` if extension is `.ts`

Reply:
I found this feature, but seems it not work when we call `transformSync`. Seems fix it in rust side is better.